multiview/cotraining/ctclassifier.py

```diff
@@ -0,0 +1,364 @@
+
+from .base import BaseCoTrainEstimator
+import numpy as np
+from sklearn.naive_bayes import GaussianNB
+from ..utils.utils import check_Xs
+from ..utils.utils import check_Xs_y_nan_allowed
+
+
+class CTClassifier(BaseCoTrainEstimator):
+    """
+    Co-Training Classifier
+
+    This class implements the co-training classifier with the framework as
+    described in [1]. This should ideally be used on 2 views of the input
+    data which satisfy the 3 conditions for multi-view co-training
+    (sufficiency, compatibility, conditional independence) as detailed in [1].
+
+    Parameters
+    ----------
+    estimator1 : classifier object
+        The classifier object which will be trained on view 1 of the data.
+        This classifier should support the predict_proba() function so that
+        classification probabilities can be computed and co-training can be
+        performed effectively.
+
+    estimator2 : classifier object
+        The classifier object which will be trained on view 2 of the data.
+        Does not need to be of the same type as estimator1, but should support
+        predict_proba().
+
+    Attributes
+    ----------
+    estimator1 : classifier object
+        The classifier used on view 1.
+
+    estimator2 : classifier object
+        The classifier used on view 2.
+
+    class_name: string
+        The name of the class: CTClassifier.
+
+    n_views_ : int
+        The number of views supported by the multi-view classifier
+
+    classes_ : array-like of shape (n_classes,)
+        Unique class labels.
+
+    n_classes_ : int
+        Number of unique classes.
+
+    p_ : int
+        Number of positive examples (second label in classes_) to pull from
+        unlabeled_pool and give "label" at each training round. When the
+        labels are 0 or 1, positive is defined as 1, and in general, positive
+        is the larger label.
+
+    n_ : int
+        Number of negative examples (second label in classes_) to pull from
+        unlabeled_pool and give "label" at each training round. when the
+        labels are 0 or 1, negative is defined as 0, and in general, negative
+        is the smaller label.
+
+    unlabeled_pool_size_ : int
+        Size of pool of unlabeled_pool examples to classify at each iteration.
+
+    num_iter_ : int
+        Maximum number of training iterations to run.
+
+    References
+    ----------
+    [1] Blum, A., & Mitchell, T. (1998, July). Combining labeled and
+    unlabeled_pool data with co-training. In Proceedings of the eleventh
+    annual conference on Computational learning theory (pp. 92-100). ACM.
+
+    """
+
+    def __init__(
+                 self,
+                 estimator1=None,
+                 estimator2=None,
+                 random_state=0
+                 ):
+
+        super().__init__(estimator1, estimator2)
+
+        # if not given, set classifiers as gaussian naive bayes estimators
+        if self.estimator1 is None:
```

Comment:
Nice default estimator here! Can you specify in documentation somewhere that this default is set?

Reply:
added "default (sklearn GaussianNB)" to the parameters documentation